astropy/convolution/src/convolve.c

```diff
@@ -159,6 +199,7 @@ FORCE_INLINE void convolve1d_boundary_none(DTYPE * const result,
 #endif
 
     const size_t _wkx = _nkx / 2;
+
 #ifdef NDEBUG
     if (!(_nx > 2*_wkx))
         return;
```

Comment:
I think the conditions below need to be conditional on _padded, no? In the padded case, it is implementable to have a kernel larger than the image. Might not make sense to do so, but that's the user's prerogative.

Reply:
I don't think so - in the padded case the kernel can't be larger than the image size since the image is by definition padded to include the kernel size (remember that nx is now the padded size)

Reply:
Right, it doesn't need to be there for the padded case since the assertion is guaranteed to pass. I'm not sure if it's really worth adding an if statement though in terms of code complexity? (I doubt it has any performance impact?)